sbc_translation/models/correspondence.py

```diff
@@ -22,22 +34,114 @@ class Correspondence(models.Model):
     ##########################################################################
     #                              ORM METHODS                               #
     ##########################################################################
+
     @api.model
     def create(self, vals):
-        """ Create a message for sending the CommKit only if the preferred
-            language is the same as the letter's language else send the letter
-            on the local translate plaforme.
-            """
-
+        """ Create a message for sending the CommKit after be translated on
+             the local translate plaforme.
+        """
         sponsorship = self.env['recurring.contract'].browse(
             vals['sponsorship_id'])
-        if sponsorship.reading_language.id == vals['original_language_id']:
-            letter = super(Correspondence, self).create(vals)
-        else:
+
+        original_lang = self.env['res.lang.compassion'].browse(
+            vals['original_language_id'])
+
+        if original_lang.translatable and original_lang.id not in sponsorship\
+                .child_id.project_id.country_id.spoken_lang_ids.mapped('id'):
             letter = super(Correspondence, self.with_context(
                 no_comm_kit=True)).create(vals)
-            self.alternative()
+            letter.send_local_translate()
+        else:
+            letter = super(Correspondence, self).create(vals)
+
         return letter
 
-    def alternative(self):
-        pass
+    def send_local_translate(self):
+        # Insert the letter in the mysql data base
+        tc = translate_connector.TranslateConnect()
+        # Send letter to local translate platform
+        text_id = tc.upsert_text(self.sponsorship_id, self)
+        translation_id = tc.upsert_translation(text_id, self)
+        tc.upsert_translation_status(translation_id)
+
+        # Transfert file on the NAS
+
+        # Retrieve configuration
+        smb_user = config.get('smb_user')
+        smb_pass = config.get('smb_pwd')
+        smb_ip = config.get('smb_ip')
+        smb_port = int(config.get('smb_port', 0))
+        if not (smb_user and smb_pass and smb_ip and smb_port):
+            raise Exception('No config SMB in file .conf')
+
+        # Copy file in the imported letter folder
+        smb_conn = SMBConnection(smb_user, smb_pass, 'openerp', 'nas')
+        if smb_conn.connect(smb_ip, smb_port):
+            file_ = BytesIO(base64.b64decode(
+                self.letter_image.with_context(
+                    bin_size=False).datas))
+            config_obj = self.env['ir.config_parameter']
+            nas_share_name = (config_obj.search(
+                [('key', '=', 'sbc_translation.nas_share_name')])[0]).value
+
+            child = self.sponsorship_id.child_id
+            sponsor = self.sponsorship_id.partner_id
+            letter_name = "_".join(
+                (child.code, sponsor.ref, str(self.id)))
```

Comment:
I would setup the name at the beginning of the method and pass the name as a parameter to the function `upsert_text` to ensure the name is the same in the database! (if we change the code we don't have to change in the two methods).